src/puppetlabs/puppetdb/command.clj

```diff
@@ -410,9 +410,24 @@
           :producer-timestamp (-> cmd :payload :producer_timestamp)}
          (when ex {:exception ex})))
 
+(defn call-with-quick-retry [num-retries f]
+  (loop [n num-retries]
+    (let [result (try+
```

Comment:
I think it makes sense to have a success indicator, but I think we want to return the results of invoking (f) as the return value of call-with-quick-retry